src/modules/commitment/cqrs/query/list-commitments.query.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@nestjs/common';
+import { IPaginationQueryArgs, IQuery, paginate } from '@common/cqrs';
+import { FindCriteria } from '@core/repository.interface';
+import { ICommitment, ICommitmentRepository, QueryCommitment } from '@domain/commitment';
+import { IPaginatedList } from '@domain/core';
+
+@Injectable()
+export class ListCommitmentsQuery implements IQuery<IPaginationQueryArgs<FindCriteria<QueryCommitment>>> {
```

Comment:
A good thing to do for this is also make a unit test for this query. Good to give you experience in that part of the code base too 🙂

Reply:
Hi Michael this is done